src/WidgetBase.ts

```diff
@@ -350,8 +352,21 @@ export class WidgetBase<P extends WidgetProperties> extends Evented implements W
 			if (widget) {
 				this._cachedVNode = widget;
 			}
+
+			this.emit({
```

Comment:
This highlights that this block could likely be written so there is only a single return statement (and a single emit).